iree/turbine/kernel/wave/templates/vanilla_attention.py

```diff
@@ -95,6 +98,12 @@ def repeat(
             k_reg = tkw.read(k, elements_per_thread=LOAD_ELEMS_PER_THREAD_QK)
             inner_acc = tkw.mma(k_reg, q_reg, imm_reg, mfma_variant[0])
             x_j = tkw.permute(inner_acc, target_shape=[B, M, K2])
+            if is_causal:
+                m_index = tkw.self_index(M, tkl.i64, elements_per_thread=1)
+                m_index = tkw.broadcast(m_index, target_shape=[M, K2])
```

Comment:
Note: I would keep the comment about M will resolve to 1 after mapping and therefore the broadcast will be legal thanks to static information carried by transformations. This is not at all trivial to understand and should be pointed out IMO.

Reply:
makes sense, done, thanks!